### PR TITLE
NDv5 tests

### PR DESCRIPTION
**Pull request by rajvi**

### CODE REVIEW

File: conf/nd96isr_v5.conf [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+#
+# This file was automatically generated by nhc-genconf
+# Sat May 6 00:05:37 UTC 2023
+#
+
+#######################################################################
+###
+### Filesystem checks
+###
+ ndv5-vm-0 || check_fs_mount_rw -t "ext4" -s "/dev/root" -f "/"
```
Comment (rafsalas19) [inline conf/nd96isr_v5.conf]: Lets change the "ndv5-vm-0" to "*"

File: conf/nd96isr_v5.conf [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+# Comments begin with '#'
+#
+# This file was automatically generated by nhc-genconf
+# Sat May 6 00:05:37 UTC 2023
+#
+
+#######################################################################
+###
+### Filesystem checks
+###
```
Comment (rafsalas19) [inline conf/nd96isr_v5.conf:14]: For this specific section (Filesystem checks), can you just use exactly the same checks that we use for the NDv4 conf file

File: customTests/azure_ib_write_bw_gdr.nhc [hunk 3]
```diff
@@ -5,12 +5,12 @@
 IB_WRITE_BW_EXE_PATH=/opt/azurehpc/test/nhc/perftest-4.5/ib_write_bw
 IB_WRITE_BW=`basename $IB_WRITE_BW_EXE_PATH`
 IB_WRITE_BW_DURATION=10
-IB_WRITE_BW_ARGS="-s $(( 1 * 1024 * 1024 )) -D ${IB_WRITE_BW_DURATION} -x 0 -F --report_gbits"
+IB_WRITE_BW_ARGS="-p 18516 -s $(( 1 * 1024 * 1024 )) -D ${IB_WRITE_BW_DURATION} -x 0 -F --report_gbits"
 SLEEP_TIME=5
 
 HOSTNAME=`hostname`
 #NUMA mapping for NDv4(A100)
-GPU_NUMA=( 1 1 0 0 3 3 2 2 )
```
Comment (rafsalas19) [inline customTests/azure_ib_write_bw_gdr.nhc:13]: Hmmm, so this test needs to be able to run on both NDv4 and NDv5
so for NDV4 this numa topology is needed GPU_NUMA=( 1 1 0 0 3 3 2 2 )
and for NDv5 this one: GPU_NUMA=( 0 0 0 0 1 1 1 1 )

We need to find a which numa topology to use based on what sku we are on.

**Review by rafsalas19 — CHANGES_REQUESTED** [review]

File: conf/nd96asr_v4.conf [hunk 4]
```diff
@@ -82,3 +82,4 @@
  * || check_nccl_allreduce_ib_loopback 18.0 10
  * || check_ib_link_flapping 6
  
+
```
Comment (jithinjosepkl) [inline conf/nd96asr_v4.conf]: intentional?

File: conf/nd96isr_v5.conf [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+ * || check_gpu_clock_throttling
+ * || check_nccl_allreduce 350.0 1 /opt/microsoft/ndv5-topo.xml
+
+
+#######################################################################
+####
+#### Additional IB checks
+####
+#  * || check_ib_bw_gdr 390
+ * || check_nccl_allreduce_ib_loopback 37.0 1 /opt/microsoft/ndv5-topo.xml
```
Comment (jithinjosepkl) [inline conf/nd96isr_v5.conf]: Any reason why we pick 37 here?

File: conf/nd96isr_v5.conf [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+#######################################################################
+####
+#### GPU checks
+####
+ * || check_nvsmi_healthmon
+ * || check_app_gpu_clocks
+ * || check_cuda_bw 52
+ * || check_gpu_ecc 20000000 10000
+ * || check_gpu_clock_throttling
+ * || check_nccl_allreduce 350.0 1 /opt/microsoft/ndv5-topo.xml
```
Comment (jithinjosepkl) [inline conf/nd96isr_v5.conf]: What's the message size that we are checking here? Depending on that, we may have to bump this up.

File: customTests/azure_nccl_allreduce.nhc [hunk 7]
```diff
@@ -2,8 +2,13 @@
 
 # Check for NVlink ssues by running NCCL allreduce.
 
+TOPOFILE=$3
+if [[ -z "$TOPOFILE" ]];then
```
Comment (jithinjosepkl) [inline customTests/azure_nccl_allreduce.nhc]: any reason we are using NDv4 topo by default? It maybe better to fail the test if topology file is not passed in.

File: customTests/azure_ib_write_bw_gdr.nhc (hunk 3 above)
Comment (rajvi) [reply customTests/azure_ib_write_bw_gdr.nhc:13]: Ok, I can pass in the sku from the conf file accordingly

File: customTests/azure_nccl_allreduce.nhc (hunk 7 above)
Comment (rafsalas19) [reply customTests/azure_nccl_allreduce.nhc]: That was an arbitrary decision. @rajvi please have the test fail instead.

File: conf/nd96isr_v5.conf (hunk 6 above)
Comment (rafsalas19) [reply conf/nd96isr_v5.conf]: 1G message. ANy suggestions?

File: conf/nd96isr_v5.conf (hunk 5 above)
Comment (rafsalas19) [reply conf/nd96isr_v5.conf]: This is based on results we saw when we manually ran the test.

File: conf/nd96isr_v5.conf (hunk 6 above)
Comment (jithinjosepkl) [reply conf/nd96isr_v5.conf]: I think we can use larger message size (8G or 16G), and have a larger threshold limit.

Comment (rafsalas19) [reply conf/nd96isr_v5.conf]: we will address

File: conf/nd96isr_v5.conf (hunk 2 above)
Comment (rajvi) [reply conf/nd96isr_v5.conf:14]: that should be fine. these checks are from the nhc autogeneration utility on the ndv5 vm

Comment (rajvi) [reply conf/nd96isr_v5.conf:14]: I tested using the same checks as ndv4 and some fail like these:
ERROR:  nhc:  Health check failed:  check_hw_physmem:  Actual RAM size (1961032660 kB) greater than maximum allowed (928858388 kB).
ERROR:  nhc:  Health check failed:  check_hw_ib:  No IB port mlx5_ib0:1 is ACTIVE (LinkUp 200 Gb/sec).

For those tests, I'll use the args like the ones in this file as they're specific to ndv5.

File: conf/nd96amsr_v4.conf [hunk 8]
```diff
@@ -104,13 +104,13 @@
  * || check_cuda_bw 24
  * || check_gpu_ecc 20000000 10000
  * || check_gpu_clock_throttling
- * || check_nccl_allreduce 228.0 10
+ * || check_nccl_allreduce 228.0 10 /opt/microsoft/ndv4-topo.xml 4G
```
Comment (moes1) [inline conf/nd96amsr_v4.conf:107]: This is large enough I think, but we could increase it by another factor of 2 if you want to have a policy of testing the largest sizes possible.

**Review by moes1 — APPROVED** [review]

**Review by rafsalas19 — APPROVED** [review]

File: conf/nd96amsr_v4.conf (hunk 8 above)
Comment (rajvi) [reply conf/nd96amsr_v4.conf:107]: Ok I think 8G was able to work. I'll have to double check so I'll increase it in another PR.

**Pull request merged**